src/main/java/org/opensearch/flowframework/workflow/Workflow.java

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+package org.opensearch.flowframework.workflow;
+
+/**
+ * Interface for the workflow setup of different building blocks.
+ */
+public interface Workflow {
+
+    /**
+     * Triggers the processing of the building block.
+     * @throws Exception if execution fails
+     */
+    void process() throws Exception;
```

Comment:
@dbwiddis need input on the interface for the building block. I envision that the every building block let's UploadModel would implement this interface something like:

```java
public class UploadModel implements Workflow {

    @Override
    public void process() {
       mlClient.uploadModel()
    }
}

Reply:
1. We probably want a return value of type `CompletableFuture<Workflow>`
2. I'm not overly fond of the name "process" here, as I'm thinking a workflow is a chaining of processes.   I'd prefer `execute()` or possibly  `doExecute()`.
3. I'm working on process chaining in a separate branch with a PR soon.  I don't expect my design to be final either, but let's not let this discussion hold up progress, we'll see the connections and merge/rename the approaches in a later refactor.

Reply:
Thanks for the suggestions. I have changed the method name to `execute` because `doExecute()` would create little confusion with `prepareRequest` -> `doExecute()`. Also, I have added the return type as `CompletableFuture<Workflow>`.

Reply:
Great!  Your comments on my PR #26 on how I can interface with your `execute()` from my code would be helpful.  OK to merge this one, we can always tweak later.